test-common/src/main/java/de/bwaldvogel/mongo/backend/AbstractBackendTest.java

```diff
@@ -143,6 +143,22 @@ public void testSimpleInsert() throws Exception {
         collection.insertOne(json("_id: 1"));
     }
 
+    @Test
+    public void testSimpleCursor() {
+        int expectedCount = 20;
+        int batchSize = 10;
+        for (int i = 0; i < expectedCount; i++) {
+            collection.insertOne(new Document("name", "testUser1"));
+        }
+        MongoCursor<Document> cursor = collection.find().batchSize(batchSize).cursor();
+        int count = 0;
+        while (cursor.hasNext()) {
+            cursor.next();
+            count++;
+        }
+        assertThat(count).isEqualTo(expectedCount);
```

Comment:
validate that cursor state is cleaned-up on server side after exhausting the iteration or explicitly calling `close()`